src/ray/core_worker/experimental_mutable_object_provider.cc

```diff
@@ -184,19 +186,24 @@ void MutableObjectProvider::PollWriterClosure(
   RAY_CHECK(object->GetData());
   RAY_CHECK(object->GetMetadata());
 
-  reader->PushMutableObject(
-      object_id,
-      object->GetData()->Size(),
-      object->GetMetadata()->Size(),
-      object->GetData()->Data(),
-      [this, &io_context, object_id, reader](const Status &status,
-                                             const rpc::PushMutableObjectReply &reply) {
-        io_context.post(
-            [this, &io_context, object_id, reader]() {
-              PollWriterClosure(io_context, object_id, reader);
-            },
-            "experimental::MutableObjectProvider.PollWriter");
-      });
+  auto counter = std::make_shared<std::atomic<int>>(readers.size());
```

Comment:
Can we use `thread_local` here? I haven't heard about this keyword before, but ChatGPT tells me that if we declare a variable as `thread_local`, it becomes independent between threads. I guess the callback function is called by different threads created by each RPC call. If counter becomes independent between threads, counter will just be `readers.size() - 1` in each thread.

Reply:
We can pass the `thread_local` variable by reference to the callback function, so it will be fine even if the callback is executed by a different thread.